apps/system/lockscreen/js/lockscreen_powerbutton_manager.js

```diff
@@ -0,0 +1,116 @@
+/**
+ Copyright 2015, Mozilla Foundation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * LockScreen's handling of the power button is implemented here.
+ */
+(function(exports) {
+
+  var LockScreenPowerbuttonManager = function() {
+    /**
+     * Boolean value to track state of locksckscreen.enabled setting.
+     * We must default to false here to ensure that a potential race condition
+     * with the settings observer never results in a permanently locked device.
+     */
+    this.lockScreenEnabled = false;
+    /** String value defining power button behavior. Currently implemented:
+     * - 'screen-lock': default behavior of just locking the screen
+     * - 'passcode-lock': locks and immediately requires pass code on unlock
+     * Only settings listener should change this value to sync with data
+     * in Settings API.
+     */
+    this.buttonBehavior = 'screen-lock';
+  };
+
+  LockScreenPowerbuttonManager.prototype.start =
+    function lspm_start() {
+      window.SettingsListener.observe(
```

Comment:
Personally I support what Tim said at

https://groups.google.com/forum/#!topic/mozilla.dev.b2g/ZPiHWOSWQR8

So unless I'm moving old code around I would avoid adding new lines using SettingListeners. If it's not necessary, could you change this and to use plain Settings API